torch/csrc/distributed/rpc/tensorpipe_agent.cpp

```diff
@@ -405,6 +413,51 @@ std::string TensorPipeAgent::createUniqueShmAddr() {
 }
 #endif
 
+std::string TensorPipeAgent::getDefaultIPAddress() {
+  char hostname[HOST_NAME_MAX];
+  int rv = gethostname(hostname, HOST_NAME_MAX);
+  TORCH_INTERNAL_ASSERT(rv == 0, "Unable to get local hostname");
+
+  struct addrinfo hints, *servinfo, *p;
+  memset(&hints, 0, sizeof hints);
+  hints.ai_family = AF_UNSPEC;
+  hints.ai_socktype = SOCK_STREAM;
```

Comment:
```suggestion
  hints.ai_socktype = SOCK_STREAM;
  hints.ai_protocol = IPPROTO_TCP;
```
Does it perhaps make sense to only ask for addresses that support TCP? I don't know if it ever makes a difference, but it shouldn't hurt either...

Reply:
Just followed Gloo and it shouldn't hurt..